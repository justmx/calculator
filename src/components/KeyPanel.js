import React from 'react';
import Button from "./Button";

const KeyPanel = ({onClick, content, left_data, right_data}) => {
  return (
     <div className='calculator'>
        <div className='display'>
          {content}
        </div>
        <div className='calculator-container'>
            <div className='flex-row-between left'>
              {
                left_data.map((item, i) => {
                  return <Button content={{content: item}} onClick={onClick} key={i} left />
                })
              }
            </div>
            <div className='flex-row-between right'>
              {
                right_data.map((item, i) => {
                  return (<Button content={item} onClick={onClick} key={i}/>)
                })
              }
            </div>
          </div>
    </div>
  )
}

export default KeyPanel;
