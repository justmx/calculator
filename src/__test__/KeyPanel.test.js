import React from 'react';
import { shallow } from 'enzyme';
import KeyPanel from '../components/KeyPanel';

const left_data = [
  '(', ')', 'mc', 'm+', 'm-', 'mr'
]

const right_data = [
{
  
  content: 'C',
  colour: '#a4a4a4'
},
{
  
  content: '+/-',
  colour: '#a4a4a4'
},
{
  
  content: '%',
  colour: '#a4a4a4'
},
{
  
  content: '/',
  colour: '#e49425'
},
{
  
  content: '7',
  colour: '#363636'
}];

describe('KeyPanel', () => {
  it('should render KeyPanel', () => {
    const keyPanel = shallow(<KeyPanel left_data={left_data} right_data={right_data}/>);
    expect(keyPanel).toMatchSnapshot();
  });

  it('renders Buttons', () => {
    const keyPanel = shallow(<KeyPanel left_data={left_data} right_data={right_data}/>);
    expect(keyPanel.find('Button')).toBeDefined();
    expect(keyPanel.find('Button')).toHaveLength(left_data.length + right_data.length);
  });
});
