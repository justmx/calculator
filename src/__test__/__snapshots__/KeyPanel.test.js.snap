// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyPanel should render KeyPanel 1`] = `
<div
  className="calculator"
>
  <div
    className="display"
  />
  <div
    className="calculator-container"
  >
    <div
      className="flex-row-between left"
    >
      <Button
        content={
          Object {
            "content": "(",
          }
        }
        key="0"
        left={true}
      />
      <Button
        content={
          Object {
            "content": ")",
          }
        }
        key="1"
        left={true}
      />
      <Button
        content={
          Object {
            "content": "mc",
          }
        }
        key="2"
        left={true}
      />
      <Button
        content={
          Object {
            "content": "m+",
          }
        }
        key="3"
        left={true}
      />
      <Button
        content={
          Object {
            "content": "m-",
          }
        }
        key="4"
        left={true}
      />
      <Button
        content={
          Object {
            "content": "mr",
          }
        }
        key="5"
        left={true}
      />
    </div>
    <div
      className="flex-row-between right"
    >
      <Button
        content={
          Object {
            "colour": "#a4a4a4",
            "content": "C",
          }
        }
        key="0"
      />
      <Button
        content={
          Object {
            "colour": "#a4a4a4",
            "content": "+/-",
          }
        }
        key="1"
      />
      <Button
        content={
          Object {
            "colour": "#a4a4a4",
            "content": "%",
          }
        }
        key="2"
      />
      <Button
        content={
          Object {
            "colour": "#e49425",
            "content": "/",
          }
        }
        key="3"
      />
      <Button
        content={
          Object {
            "colour": "#363636",
            "content": "7",
          }
        }
        key="4"
      />
    </div>
  </div>
</div>
`;
